harness/determined/core/_train.py

```diff
@@ -89,10 +112,55 @@ def report_training_metrics(
             data=det.util.json_encode(body),
         )
 
-        # Also sync tensorboard.
-        if self._tbd_writer and self._tbd_mgr:
+        if self._tbd_sync_mode == TensorboardSyncMode.Auto:
+            self.write_training_tensorboard_metrics(steps_completed, metrics, batch_metrics)
+            if self._tbd_mgr:
+                self._tbd_mgr.sync()
+
+    def write_training_tensorboard_metrics(
+        self,
+        steps_completed: int,
+        metrics: Dict[str, Any],
+        batch_metrics: Optional[List[Dict[str, Any]]] = None,
+    ) -> None:
+        if self._tbd_writer:
             self._tbd_writer.on_train_step_end(steps_completed, metrics, batch_metrics)
```

Comment:
hm... I know I thought this `write_training_tensorboard_metrics()` was a good idea, but now that I see it, I think it's a rather bad idea.  The calling pattern here would be:

- user creates core api with manual tb mode
- user creates one of our tensorboard.MetricWriter objects, which is our own internal interface, wrapped in a BatchMetricWriter, another one of our internal interfaces.
- user passes the BatchMetricWriter to set_tensorboard_metric_writer()
- user then is able to call write_training_tensorboard_metrics()
- write_training_tensorboard_metrics() just passes its args to the BatchMetricWriter that they already created
- user calls upload_tensorboard_metrics() to sync the tensorboard to persistent storage


This is pretty complex.  I'd rather the manual tb calling pattern look like:

- user creates core api with manual tb mode
- user writes metrics literally however they usually write metrics (not using our internal MetricWriter interfaces)
- user calls upload_tensorboard_metrics() to sync the tensorboard to persistent storage